x-pack/plugins/cases/server/connectors/cases/cases_connector_executor.test.ts

```diff
@@ -0,0 +1,2086 @@
+          rule,
+          timeWindow,
+          reopenClosedCases,
+        });
+
+        expect(casesClientMock.attachments.bulkCreate).toHaveBeenCalledTimes(1);
+
+        expect(casesClientMock.attachments.bulkCreate).nthCalledWith(1, {
+          caseId: 'mock-id-1',
+          attachments: alerts.map((alert) => ({
+            alertId: alert._id,
+            index: alert._index,
+            owner: 'securitySolution',
+            rule: {
+              id: 'rule-test-id',
+              name: 'Test rule',
+            },
+            type: 'alert',
+          })),
+        });
+      });
+    });
+  });
+
+  describe('Retries', () => {
+    it('attaches the alerts correctly when bulkGetRecords fails', async () => {
+      mockBulkGetRecords
+        .mockResolvedValueOnce([
+          {
+            id: groupedAlertsWithOracleKey[2].oracleKey,
+            type: CASE_ORACLE_SAVED_OBJECT,
+            message: 'getting records: mockBulkGetRecords error',
+            statusCode: 409,
+            error: 'Conflict',
+          },
+        ])
+        .mockResolvedValueOnce(oracleRecords);
+
+      await expect(() =>
+        connectorExecutor.execute({
+          alerts,
+          groupingBy,
+          owner,
+          rule,
+          timeWindow,
+          reopenClosedCases,
+        })
+      ).rejects.toThrowErrorMatchingInlineSnapshot(
+        `"Conflict: getting records: mockBulkGetRecords error"`
+      );
+
+      resetCounters();
+
+      // retry
+      await connectorExecutor.execute({
+        alerts,
+        groupingBy,
+        owner,
+        rule,
+        timeWindow,
+        reopenClosedCases,
+      });
+
+      expectCasesToHaveTheCorrectAlertsAttachedWithGrouping(casesClientMock);
+    });
+
+    it('attaches the alerts correctly when bulkCreateRecord fails', async () => {
+      mockBulkCreateRecords
+        .mockResolvedValueOnce([
+          {
+            id: groupedAlertsWithOracleKey[2].oracleKey,
+            type: CASE_ORACLE_SAVED_OBJECT,
+            message: 'creating records: bulkCreateRecord error',
+            statusCode: 409,
+            error: 'Conflict',
+          },
+        ])
+        .mockResolvedValueOnce([createdOracleRecord]);
+
+      await expect(() =>
+        connectorExecutor.execute({
+          alerts,
+          groupingBy,
+          owner,
+          rule,
+          timeWindow,
+          reopenClosedCases,
+        })
+      ).rejects.toThrowErrorMatchingInlineSnapshot(
+        `"Conflict: creating records: bulkCreateRecord error"`
+      );
+
+      resetCounters();
+
+      // retry
+      await connectorExecutor.execute({
+        alerts,
+        groupingBy,
+        owner,
+        rule,
+        timeWindow,
+        reopenClosedCases,
+      });
+
+      expectCasesToHaveTheCorrectAlertsAttachedWithGrouping(casesClientMock);
+    });
+
+    it('attaches the alerts correctly while creating a record and another node has already created it', async () => {
```

Comment:
Isn't this fundamentally the same as `'attaches the alerts correctly when bulkCreateRecord fails'`?